core/shared/src/test/scala/cats/effect/std/internal/BinomialHeapSpec.scala

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2020 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * These tests have been inspired by and adapted from `monix-catnap`'s `ConcurrentQueueSuite`, available at
+ * https://github.com/monix/monix/blob/series/3.x/monix-catnap/shared/src/test/scala/monix/catnap/ConcurrentQueueSuite.scala.
+ */
+
+package cats.effect.std.internal
+
+import cats.Order
+
+import org.specs2.mutable.Specification
+import org.specs2.ScalaCheck
+
+import org.scalacheck.{Arbitrary, Gen}, Arbitrary.arbitrary
+
+class BinomialHeapSpec extends Specification with ScalaCheck {
+
+  implicit val orderForInt: Order[Int] = Order.fromLessThan((x, y) => x < y)
+
+  "Binomial heap" should {
+
+    "dequeue by priority" in prop { elems: List[Int] =>
```

Comment:
I'm actually surprised this compiles as-is on dotty. Something rather strange is probably going on.

```suggestion
    "dequeue by priority" in prop { (elems: List[Int]) =>
```

And similarly for all such lambdas.